WIP: support table multi data base on vllm-0.5.4

oat-mirror opened the pull request:
TODO:

- [ ]  Simplify the code by removing redundant sections.
- [ ]  Modify the architecture and model types to align with `TableGPT`
- [ ]  Adding some auto-mapping files, such as `modeling_tablegpt.py`, and related files for running the model with transformers.
- [ ]  migrate to the version based on vLLM 0.5.5

youkaichao committed:
[ci][frontend] deduplicate tests (#7101)

tdoublep committed:
[Doc] [SpecDecode] Update MLPSpeculator documentation (#7100)

Signed-off-by: Thomas Parnell <tpa@zurich.ibm.com>

jischein committed:
[Bugfix] Specify device when loading LoRA and embedding tensors (#7129)

Co-authored-by: Jacob Schein <jacobschein@Jacobs-MacBook-Pro-2.local>

comaniac committed:
[MISC] Use non-blocking transfer in prepare_input (#7172)

Isotr0py committed:
[Core] Support loading GGUF model (#5191)

Co-authored-by: Michael Goin <michael@neuralmagic.com>

simon-mo committed:
[Build] Add initial conditional testing spec (#6841)

jeejeelee committed:
[LoRA] Relax LoRA condition (#7146)

DarkLight1337 committed:
[Model] Support SigLIP encoder and alternative decoders for LLaVA models (#7153)

Co-authored-by: Roger Wang <136131678+ywang96@users.noreply.github.com>

dsikka committed:
[BugFix] Fix DeepSeek remote code (#7178)

robertgshaw2-redhat committed:
[ BugFix ] Fix ZMQ when `VLLM_PORT` is set (#7205)

kpapis committed:
[Bugfix] add gguf dependency (#7198)

Co-authored-by: katarzyna.papis <kpapis@kpapis-u20.sclab.intel.com>

LiuXiaoxuanPKU committed:
[SpecDecode] [Minor] Fix spec decode sampler tests (#7183)

ProExpertProg committed:
[Kernel] Add per-tensor and per-token AZP epilogues (#5941)

Co-authored-by: Tyler Michael Smith <tyler@neuralmagic.com>

xiaobochen123 committed:
[Core] Optimize evictor-v2 performance (#7193)

afeldman-nm committed:
[Core] Subclass ModelRunner to support cross-attention & encoder sequences (towards eventual encoder/decoder model support) (#4942)

Co-authored-by: Andrew Feldman <afeld2012@gmail.com>
Co-authored-by: Nick Hill <nickhill@us.ibm.com>

mgoin committed:
[Bugfix] Fix GPTQ and GPTQ Marlin CPU Offloading (#7225)

njhill committed:
[BugFix] Overhaul async request cancellation (#7111)

ywang96 committed:
[Doc] Mock new dependencies for documentation (#7245)

Atllkks10 committed:
[BUGFIX]: top_k is expected to be an integer. (#7227)

DarkLight1337 committed:
[Frontend] Gracefully handle missing chat template and fix CI failure (#7238)

Co-authored-by: Roger Wang <ywang@roblox.com>

youkaichao committed:
[distributed][misc] add specialized method for cuda platform (#7249)

dsikka committed:
[Misc] Refactor linear layer weight loading; introduce `BasevLLMParameter` and `weight_loader_v2` (#5874)

robertgshaw2-redhat committed:
[ BugFix ] Move `zmq` frontend to IPC instead of TCP (#7222)

rafvasq committed:
Fixes typo in function name (#7275)

Signed-off-by: Rafael Vasquez <rafvasq21@gmail.com>

Isotr0py committed:
[Bugfix] Fix input processor for InternVL2 model (#7164)

Co-authored-by: Cyrus Leung <cyrus.tl.leung@gmail.com>

ilya-lavrenov committed:
[OpenVINO] migrate to latest dependencies versions (#7251)

stas00 committed:
[Doc] add online speculative decoding example (#7243)

maxdebayser committed:
[BugFix] Fix frontend multiprocessing hang (#7217)

Signed-off-by: Max de Bayser <mbayser@br.ibm.com>
Co-authored-by: Robert Shaw <114415538+robertgshaw2-neuralmagic@users.noreply.github.com>

mgoin committed:
[Bugfix][FP8] Fix dynamic FP8 Marlin quantization (#7219)

khluu committed:
[ci] Make building wheels per commit optional (#7278)

Signed-off-by: kevin <kevin@anyscale.com>

LucasWilkinson committed:
[Bugfix] Fix gptq failure on T4s (#7264)

njhill committed:
[FrontEnd] Make `merge_async_iterators` `is_cancelled` arg optional (#7282)

mgoin committed:
[Doc] Update supported_hardware.rst (#7276)

LiuXiaoxuanPKU committed:
[Kernel] Fix Flashinfer Correctness (#7284)

ruisearch42 committed:
[Misc] Fix typos in scheduler.py (#7285)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>

NiuBlibing committed:
[Frontend] remove max_num_batched_tokens limit for lora (#7288)

andoorve committed:
[Bugfix] Fix LoRA with PP (#7292)

jeejeelee committed:
[Model] Rename MiniCPMVQwen2 to MiniCPMV2.6 (#7273)

ProExpertProg committed:
[Bugfix][Kernel] Increased atol to fix failing tests (#7305)

joerunde committed:
[Frontend] Kill the server on engine death (#6594)

Signed-off-by: Joe Runde <joe@joerun.de>
Signed-off-by: Joe Runde <Joseph.Runde@ibm.com>

zachzzc committed:
[Bugfix][fast] Fix the get_num_blocks_touched logic (#6849)

mgoin committed:
[Doc] Put collect_env issue output in a <detail> block (#7310)

dtrifiro committed:
[CI/Build] Dockerfile.cpu improvements (#7298)

Isotr0py committed:
[Bugfix] Fix new Llama3.1 GGUF model loading (#7269)

jeejeelee committed:
[Misc] Temporarily resolve the error of BitAndBytes (#7308)

simon-mo committed:
Add Skywork AI as Sponsor (#7314)

lsy323 committed:
[TPU] Add Load-time W8A16 quantization for TPU Backend (#7005)

DarkLight1337 committed:
[Core] Support serving encoder/decoder models (#7258)

WoosukKwon committed:
[TPU] Fix dockerfile.tpu (#7331)

alexm-redhat committed:
[Performance] Optimize e2e overheads: Reduce python allocations (#7162)

tjohnson31415 committed:
[Bugfix] Fix speculative decoding with MLPSpeculator with padded vocabulary (#7218)

Signed-off-by: Travis Johnson <tsjohnso@us.ibm.com>

SolitaryThinker committed:
[Speculative decoding] [Multi-Step] decouple should_modify_greedy_probs_inplace (#6971)

njhill committed:
[Core] Streamline stream termination in `AsyncLLMEngine` (#7336)

mzusman committed:
[Model][Jamba] Mamba cache single buffer  (#6739)

Co-authored-by: Mor Zusman <morz@ai21.com>

Isotr0py committed:
[VLM][Doc] Add `stop_token_ids` to InternVL example (#7354)

alexm-redhat committed:
[Performance] e2e overheads reduction: Small followup diff (#7364)

alexm-redhat committed:
[Bugfix] Fix reinit procedure in ModelInputForGPUBuilder (#7360)

pooyadavoodi committed:
[Frontend] Support embeddings in the run_batch API (#7132)

Co-authored-by: Simon Mo <simon.mo@hey.com>

ywang96 committed:
[Bugfix] Fix ITL recording in serving benchmark (#7372)

sfc-gh-mkeralapura committed:
[Core] Add span metrics for model_forward, scheduler and sampler time (#7089)

dsikka committed:
[Bugfix] Fix `PerTensorScaleParameter` weight loading for fused models (#7376)

Yard1 committed:
[Misc] Add numpy implementation of `compute_slot_mapping` (#7377)

cadedaniel committed:
[Core] Fix edge case in chunked prefill + block manager v2 (#7380)

Isotr0py committed:
[Bugfix] Fix phi3v batch inference when images have different aspect ratio (#7392)

WoosukKwon committed:
[TPU] Use mark_dynamic to reduce compilation time (#7340)

noamgat committed:
Updating LM Format Enforcer version to v0.10.6 (#7189)

SolitaryThinker committed:
[core] [2/N] refactor worker_base input preparation for multi-step (#7387)

tomeras91 committed:
[CI/Build] build on empty device for better dev experience (#4773)

tomeras91 committed:
[Doc] building vLLM with VLLM_TARGET_DEVICE=empty (#7403)

youkaichao committed:
[misc] add commit id in collect env (#7405)

simon-mo committed:
[Docs] Update readme (#7316)

ywang96 committed:
[CI/Build] Minor refactoring for vLLM assets (#7407)

LiuXiaoxuanPKU committed:
[Kernel] Flashinfer correctness fix for v0.1.3 (#7319)

ywang96 committed:
[Core][VLM] Support image embeddings as input (#6613)

DarkLight1337 committed:
[Frontend] Disallow passing `model` as both argument and option (#7347)

dtrifiro committed:
[CI/Build] bump Dockerfile.neuron image base, use public ECR (#6832)

WoosukKwon committed:
[Bugfix] Fix logit soft cap in flash-attn backend (#7425)

khluu committed:
[ci] Entrypoints run upon changes in vllm/ (#7423)

Signed-off-by: kevin <kevin@anyscale.com>

khluu committed:
[ci] Cancel fastcheck run when PR is marked ready (#7427)

Signed-off-by: kevin <kevin@anyscale.com>

khluu committed:
[ci] Cancel fastcheck when PR is ready (#7433)

Signed-off-by: kevin <kevin@anyscale.com>

LucasWilkinson committed:
[Misc] Use scalar type to dispatch to different `gptq_marlin` kernels (#7323)

DarkLight1337 committed:
[Core] Consolidate `GB` constant and enable float GB arguments (#7416)

jon-chuang committed:
[Core/Bugfix] Add FP8 K/V Scale and dtype conversion for prefix/prefill Triton Kernel (#7208)

Co-authored-by: Cody Yu <hao.yu.cody@gmail.com>

sasha0552 committed:
[Bugfix] Handle PackageNotFoundError when checking for xpu version (#7398)

dtrifiro committed:
[CI/Build] bump minimum cmake version (#6999)

ruisearch42 committed:
[Core] Shut down aDAG workers with clean async llm engine exit (#7224)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>

DarkLight1337 committed:
[mypy] Misc. typing improvements (#7417)

aw632 committed:
[Misc] improve logits processors logging message (#7435)

khluu committed:
[ci] Remove fast check cancel workflow (#7455)

DarkLight1337 committed:
[Bugfix] Fix weight loading for Chameleon when TP>1 (#7410)

youkaichao committed:
[hardware] unify usage of is_tpu to current_platform.is_tpu() (#7102)

WoosukKwon committed:
[TPU] Suppress import custom_ops warning (#7458)

WoosukKwon committed:
Revert "[Doc] Update supported_hardware.rst (#7276)" (#7467)

petersalas committed:
[Frontend][Core] Add plumbing to support audio language models (#7446)

dsikka committed:
[Misc] Update LM Eval Tolerance (#7473)

dsikka committed:
[Misc] Update `gptq_marlin` to use new vLLMParameters (#7281)

dsikka committed:
[Misc] Update Fused MoE weight loading (#7334)

dsikka committed:
[Misc] Update `awq` and `awq_marlin` to use `vLLMParameters` (#7422)

simon-mo committed:
Announce NVIDIA Meetup (#7483)

Co-authored-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

youkaichao committed:
[frontend] spawn engine process from api server process (#7484)

kylesayrs committed:
[Misc] `compressed-tensors` code reuse (#7277)

youkaichao committed:
[misc][plugin] add plugin system implementation (#7426)

WoosukKwon committed:
[TPU] Support multi-host inference (#7457)

WoosukKwon committed:
[Bugfix][CI] Import ray under guard (#7486)

jeejeelee committed:
[CI/Build]Reduce the time consumption for LoRA tests (#7396)

youkaichao committed:
[misc][ci] fix cpu test with plugins (#7489)

DarkLight1337 committed:
[Bugfix][Docs] Update list of mock imports (#7493)

youkaichao committed:
[doc] update test script to include cudagraph (#7501)

CatherineSue committed:
Fix empty output when temp is too low (#2937)

Co-authored-by: Cyrus Leung <tlleungac@connect.ust.hk>

youkaichao committed:
[ci] fix model tests (#7507)

QwertyJack committed:
[Bugfix][Frontend] Disable embedding API for chat models (#7504)

Co-authored-by: jack <jack@alex>

wallashss committed:
[Misc] Deprecation Warning when setting --engine-use-ray (#7424)

Signed-off-by: Wallas Santos <wallashss@ibm.com>
Co-authored-by: youkaichao <youkaichao@gmail.com>
Co-authored-by: Nick Hill <nickhill@us.ibm.com>
Co-authored-by: youkaichao <youkaichao@126.com>

DarkLight1337 committed:
[VLM][Core] Support profiling with multiple multi-modal inputs per prompt (#7126)

SolitaryThinker committed:
[core] [3/N] multi-step args and sequence.py (#7452)

WoosukKwon committed:
[TPU] Set per-rank XLA cache (#7533)

kylesayrs committed:
[Misc] Revert `compressed-tensors` code reuse (#7521)

pavanjava committed:
llama_index serving integration documentation (#6973)

Co-authored-by: pavanmantha <pavan.mantha@thevaslabs.io>

WoosukKwon committed:
[Bugfix][TPU] Correct env variable for XLA cache path (#7544)

omrishiv committed:
[Bugfix] update neuron for version > 0.5.0 (#7175)

Signed-off-by: omrishiv <327609+omrishiv@users.noreply.github.com>
Co-authored-by: Cyrus Leung <cyrus.tl.leung@gmail.com>

philo-he committed:
[Misc] Update dockerfile for CPU to cover protobuf installation (#7182)

mgoin committed:
[Bugfix] Fix default weight loading for scalars (#7534)

mgoin committed:
[Bugfix][Harmless] Fix float16 dtype for model_is_embedding (#7566)

ShangmingCai committed:
[Misc] Add quantization config support for speculative model. (#7343)

gnpinkert committed:
[Feature]: Add OpenAI server prompt_logprobs support #6508 (#7453)

youkaichao committed:
[ci/test] rearrange tests and make adag test soft fail (#7572)

nunjunj committed:
Chat method for offline llm (#5049)

Co-authored-by: nunjunj <ray@g-3ff9f30f2ed650001.c.vllm-405802.internal>
Co-authored-by: nunjunj <ray@g-1df6075697c3f0001.c.vllm-405802.internal>
Co-authored-by: nunjunj <ray@g-c5a2c23abc49e0001.c.vllm-405802.internal>
Co-authored-by: Cyrus Leung <cyrus.tl.leung@gmail.com>
Co-authored-by: DarkLight1337 <tlleungac@connect.ust.hk>

mgoin committed:
[CI] Move quantization cpu offload tests out of fastcheck (#7574)

jon-chuang committed:
[Misc/Testing] Use `torch.testing.assert_close` (#7324)

youkaichao committed:
register custom op for flash attn and use from torch.ops (#7536)

njhill committed:
[Core] Use uvloop with zmq-decoupled front-end (#7570)

KuntaiDu committed:
[CI] Fix crashes of performance benchmark  (#7500)

gongdao123 committed:
[Bugfix][Hardware][AMD][Frontend] add quantization param to embedding checking method (#7513)

fzyzcjy committed:
support tqdm in notebooks (#7510)

charlifu committed:
[Feature][Hardware][Amd] Add fp8 Linear Layer for Rocm (#7210)

mzusman committed:
[Kernel] W8A16 Int8 inside FusedMoE  (#7415)

mgoin committed:
[Kernel] Add tuned triton configs for ExpertsInt8 (#7601)

SolitaryThinker committed:
[spec decode] [4/N] Move update_flash_attn_metadata to attn backend (#7571)

Co-authored-by: Cody Yu <hao.yu.cody@gmail.com>

sfc-gh-mkeralapura committed:
[Core] Fix tracking of model forward time in case of PP>1 (#7440)

[Core] Fix tracking of model forward time to the span traces in case of PP>1 (#7440)

mgoin committed:
[Doc] Add docs for llmcompressor INT8 and FP8 checkpoints (#7444)

mgoin committed:
[Doc] Update quantization supported hardware table (#7595)

bnellnm committed:
[Kernel] register punica functions as torch ops (#7591)

bnellnm committed:
[Kernel][Misc] dynamo support for ScalarType (#7594)

bnellnm committed:
[Kernel] fix types used in aqlm and ggml kernels to support dynamo (#7596)

mgoin committed:
[Model] Align nemotron config with final HF state and fix lm-eval-small (#7611)

bnellnm committed:
[Bugfix] Fix custom_ar support check (#7617)

Alexei-V-Ivanov-AMD committed:
.[Build/CI] Enabling passing AMD tests. (#7610)

ruisearch42 committed:
[Bugfix] Clear engine reference in AsyncEngineRPCServer (#7618)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>

rkooo567 committed:
[aDAG] Unflake aDAG + PP tests (#7600)

metasyn committed:
[Bugfix] add >= 1.0 constraint for openai dependency (#7612)

youkaichao committed:
[misc] use nvml to get consistent device name (#7582)

youkaichao committed:
[ci][test] fix engine/logger test (#7621)

youkaichao committed:
[core][misc] update libcudart finding (#7620)

Co-authored-by: cjackal <44624812+cjackal@users.noreply.github.com>

mrbesher committed:
[Model] Pipeline parallel support for JAIS (#7603)

youkaichao committed:
[ci][test] allow longer wait time for api server (#7629)

jeejeelee committed:
[Misc]Fix BitAndBytes exception messages (#7626)

ywang96 committed:
[VLM] Refactor `MultiModalConfig` initialization and profiling (#7530)

WoosukKwon committed:
[TPU] Skip creating empty tensor (#7630)

WoosukKwon committed:
[TPU] Use mark_dynamic only for dummy run (#7634)

WoosukKwon committed:
[TPU] Optimize RoPE forward_native2 (#7636)

robertgshaw2-redhat committed:
[ Bugfix ] Fix Prometheus Metrics With `zeromq` Frontend (#7279)

Co-authored-by: Nick Hill <nickhill@us.ibm.com>

alex-jw-brooks committed:
[CI/Build] Add text-only test for Qwen models  (#7475)

Signed-off-by: Alex-Brooks <Alex.Brooks@ibm.com>

WoosukKwon committed:
[Misc] Refactor Llama3 RoPE initialization (#7637)

rkooo567 committed:
[Core] Optimize SPMD architecture with delta + serialization optimization (#7109)

peng1999 committed:
[Core] Use flashinfer sampling kernel when available (#7137)

Co-authored-by: Michael Goin <michael@neuralmagic.com>

jikunshang committed:
[Bugfix] Fix xpu build (#7644)

WoosukKwon committed:
[Misc] Remove Gemma RoPE (#7638)

comaniac committed:
[MISC] Add prefix cache hit rate to metrics (#7606)

c3-ali committed:
[Bugfix] fix lora_dtype value type in arg_utils.py - part 2 (#5428)

SolitaryThinker committed:
[core] Multi Step Scheduling (#7000)

Co-authored-by: afeldman-nm <156691304+afeldman-nm@users.noreply.github.com>

Isotr0py committed:
[Core] Support tensor parallelism for GGUF quantization (#7520)

a-ys committed:
[Bugfix] Don't disable existing loggers (#7664)

WoosukKwon committed:
[TPU] Remove redundant input tensor cloning (#7660)

tjohnson31415 committed:
[Bugfix] use StoreBoolean instead of type=bool for --disable-logprobs-during-spec-decoding (#7665)

Signed-off-by: Travis Johnson <tsjohnso@us.ibm.com>

youkaichao committed:
[doc] fix doc build error caused by msgspec (#7659)

abhigoyal1997 committed:
[Speculative Decoding] Fixing hidden states handling in batch expansion (#7508)

khluu committed:
[ci] Install Buildkite test suite analysis (#7667)

Signed-off-by: kevin <kevin@anyscale.com>

zijian-hu committed:
[Bugfix] support `tie_word_embeddings` for all models (#5724)

edwardzjl commented on vllm/entrypoints/chat_utils.py:
```diff
@@ -122,6 +123,108 @@ def _get_full_image_text_prompt(image_token_str: str, text_prompt: str) -> str:
+    :param comments: 列名的备注信息, dict
+    :return: 能反馈 dataframe 的信息
+    """
+    # from config import INSERT_SEP_TOKEN, INSERT_EMBS_TOKEN
```
Cleanup these comment lines.

edwardzjl commented on vllm/entrypoints/chat_utils.py:
```diff
@@ -112,8 +115,6 @@ def _image_token_str(model_config: ModelConfig,
     raise TypeError(f"Unknown model type: {model_type}")
 
 
-# TODO: Let user specify how to insert image tokens into prompt
```
Leave these lines untouched?

edwardzjl commented on vllm/entrypoints/chat_utils.py:
```diff
@@ -122,6 +123,108 @@ def _get_full_image_text_prompt(image_token_str: str, text_prompt: str) -> str:
     return f"{image_token_str}\n{text_prompt}"
 
 
+def __dataframe_info_simple(df:pd.DataFrame, df_name:str, comments=None):
```
profile this function, see how much time it needs.

edwardzjl replied:
format params

edwardzjl commented on vllm/entrypoints/chat_utils.py:
```diff
@@ -122,6 +123,108 @@ def _get_full_image_text_prompt(image_token_str: str, text_prompt: str) -> str:
+
+
+# df, file name without suffix
+def __build_table_question(dfs:list[pd.DataFrame],query:str,comments=None):
```
`-> str`

edwardzjl commented on vllm/entrypoints/chat_utils.py:
```diff
@@ -122,6 +123,108 @@ def _get_full_image_text_prompt(image_token_str: str, text_prompt: str) -> str:
+    """
+    # from config import INSERT_SEP_TOKEN, INSERT_EMBS_TOKEN
+    # df_info_template_simple = """/*\nDetails about the '{df_name}' dataframe include the data types, comments, the column values info as follows:\n{desc_info}\n*/"""
+    df_info_template_simple = """/*\nDetails about the '{df_name}' dataframe that can be used as follows:\n{desc_info}\n*/"""
```
Replace '\n' with an actual newline ('\n') in triple-quoted strings.

edwardzjl commented on vllm/entrypoints/chat_utils.py:
````diff
@@ -122,6 +123,108 @@ def _get_full_image_text_prompt(image_token_str: str, text_prompt: str) -> str:
+
+    """
+    
+    pref = '''With several pandas dataframes available, your task is to write the Python code to address the user's question.\n\n## Follow this format:\nQuestion: The user's query.\nThought: Evaluate the dataframes and the question to determine the solution.\nPython code: Generate the Python code, within ```python ... ```.\n\n## Details about the dataframes:\n\n'''    
````
Replace '\n' with an actual newline ('\n') in triple-quoted strings.

edwardzjl commented on vllm/entrypoints/chat_utils.py:
```diff
@@ -151,23 +255,40 @@ def _parse_chat_message_content_parts(
 
     text_prompt = "\n".join(texts)
-
+    
```
Remove whitespaces.

edwardzjl commented on vllm/envs.py:
```diff
@@ -52,6 +52,17 @@
 
+    # table configs
+    VLLM_TABLE_READ_NROWS: int = 500
+    VLLM_TABLE_MAX_COL: int = 100
```
MAX_COLS

edwardzjl commented on vllm/envs.py:
```diff
@@ -52,6 +52,17 @@
+    # table configs
+    VLLM_TABLE_READ_NROWS: int = 500
+    VLLM_TABLE_MAX_COL: int = 100
+    VLLM_TABLE_MAX_ROW: int =  50
```
MAX_ROWS.

BTW, what is MAX_ROWS given we have READ_NROWS?

edwardzjl commented on vllm/envs.py:
```diff
@@ -52,6 +52,17 @@
+    VLLM_TABLE_MAX_ROW: int =  50
+
+
+    VLLM_TABLE_INSERT_EMBS_TOKEN: str = "<insert_embs>"
```
Can't we get these special tokens from tokenizer_config.json?

edwardzjl commented on vllm/multimodal/utils.py:
```diff
@@ -68,6 +69,34 @@ async def async_get_and_parse_image(image_url: str) -> MultiModalDataDict:
 
+def load_tables(table_urls: List[str], nrows) -> List[pd.DataFrame]:
+    """
+    Asynchronously load a csv file to pd.DataFrame from a Http urls or file paths
```
Is it async?

KuntaiDu committed:
[CI] Organizing performance benchmark files (#7616)

youkaichao committed:
[misc] add nvidia related library in collect env (#7674)

jianyizh committed:
[XPU] fallback to native implementation for xpu custom op (#7670)

youkaichao committed:
[misc][cuda] add warning for pynvml user (#7675)

jikunshang committed:
[Core] Refactor executor classes for easier inheritance (#7673)

[Core] Refactor executor classes to make it easier to inherit GPUExecutor (#7673)

LucasWilkinson committed:
[Kernel] (1/N) Machete - Hopper Optimized Mixed Precision Linear Kernel  (#7174)

ilya-lavrenov committed:
[OpenVINO] Updated documentation (#7687)

Isotr0py committed:
[VLM][Model] Add test for InternViT vision encoder (#7409)

jikunshang committed:
[Hardware] [Intel GPU]  refactor xpu worker/executor (#7686)

ronensc committed:
[CI/Build] Pin OpenTelemetry versions and make errors clearer (#7266)

[CI/Build] Pin OpenTelemetry versions and make a availability errors clearer (#7266)

LucasWilkinson committed:
[Misc] Add jinja2 as an explicit build requirement (#7695)

Yard1 committed:
[Core] Add `AttentionState` abstraction (#7663)

jikunshang committed:
[Intel GPU] fix xpu not support punica kernel (which use torch.library.custom_op) (#7685)

youkaichao committed:
[ci][test] adjust max wait time for cpu offloading test (#7709)

Yard1 committed:
[Core] Pipe `worker_class_fn` argument in Executor (#7707)

youkaichao committed:
[ci] try to log process using the port to debug the port usage (#7711)

Isotr0py committed:
[Model] Add AWQ quantization support for InternVL2 model (#7187)

ywang96 committed:
[Doc] Section for Multimodal Language Models (#7719)

DarkLight1337 committed:
[mypy] Enable following imports for entrypoints (#7248)

Co-authored-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>
Co-authored-by: Fei <dfdfcai4@gmail.com>

edwardzjl commented on vllm/entrypoints/chat_utils.py:
```diff
@@ -33,7 +36,20 @@ class CustomChatCompletionContentPartParam(TypedDict, total=False):
 
+class TableURL(TypedDict, total=False):
+    url: Required[List[str]]
+    """Either a URL of the image or the base64 encoded image data."""
```
i suppose you mean 'table' here?

sasha0552 committed:
[Bugfix] Mirror jinja2 in pyproject.toml (#7723)

njhill committed:
[BugFix] Avoid premature async generator exit and raise all exception variations (#7698)

learninmou committed:
[BUG] fix crash on flashinfer backend with cudagraph disabled, when attention group_size not in [1,2,4,8] (#7509)

Isotr0py committed:
[Bugfix][Hardware][CPU] Fix `mm_limits` initialization for CPU backend (#7735)

njhill committed:
[Spec Decoding] Use target model max length as default for draft model (#7706)

brian14708 committed:
[Bugfix] chat method add_generation_prompt param (#7734)

robertgshaw2-redhat committed:
[Bugfix][Frontend] Fix Issues Under High Load With `zeromq` Frontend (#7394)

Co-authored-by: Nick Hill <nickhill@us.ibm.com>

sasha0552 committed:
[Bugfix] Pass PYTHONPATH from setup.py to CMake (#7730)

SolitaryThinker committed:
[multi-step] Raise error if not using async engine (#7703)

robertgshaw2-redhat committed:
[Frontend] Improve Startup Failure UX (#7716)

SolitaryThinker committed:
[misc] Add Torch profiler support (#7451)

Co-authored-by: Cody Yu <hao.yu.cody@gmail.com>

petersalas committed:
[Model] Add UltravoxModel and UltravoxConfig (#7615)

SolitaryThinker committed:
[ci] [multi-step] narrow multi-step test dependency paths (#7760)

dsikka committed:
[Kernel]  Expand MoE weight loading + Add Fused Marlin MoE Kernel (#7527)

Co-authored-by: ElizaWszola <eliza@neuralmagic.com>

youkaichao committed:
[distributed][misc] error on same VLLM_HOST_IP setting (#7756)

gshtras committed:
[AMD][CI/Build] Disambiguation of the function call for ROCm 6.2 headers compatibility (#7477)

Co-authored-by: Charlie Fu <Charlie.Fu@amd.com>

ProExpertProg committed:
[Kernel] Replaced `blockReduce[...]` functions with `cub::BlockReduce` (#7233)

Co-authored-by: Michael Goin <michael@neuralmagic.com>

zifeitong committed:
[Model] Fix Phi-3.5-vision-instruct 'num_crops' issue (#7710)

joerunde committed:
[Bug][Frontend] Improve ZMQ client robustness (#7443)

Signed-off-by: Joe Runde <Joseph.Runde@ibm.com>

mgoin committed:
Revert "[Kernel]  Expand MoE weight loading + Add Fused Marlin MoE Kernel (#7527)" (#7764)

WoosukKwon committed:
[TPU] Avoid initializing TPU runtime in is_tpu (#7763)

youkaichao committed:
[ci] refine dependency for distributed tests (#7776)

WoosukKwon committed:
[Misc] Use torch.compile for GemmaRMSNorm (#7642)

abhigoyal1997 committed:
[Speculative Decoding] EAGLE Implementation with Top-1 proposer (#6830)

sfc-gh-zhwang committed:
Fix ShardedStateLoader for vllm fp8 quantization (#7708)

LucasWilkinson committed:
[Bugfix] Don't build machete on cuda <12.0 (#7757)

dsikka committed:
[Misc] update fp8 to use `vLLMParameter` (#7437)

tjohnson31415 committed:
[Bugfix] spec decode handle None entries in topk args in create_sequence_group_output (#7232)

Signed-off-by: Travis Johnson <tsjohnso@us.ibm.com>

Jeffwan committed:
[Misc] Enhance prefix-caching benchmark tool (#6568)

petersalas committed:
[Doc] Fix incorrect docs from #7615 (#7788)

mgoin committed:
[Bugfix] Use LoadFormat values for `vllm serve --load-format` (#7784)

khluu committed:
[ci] Cleanup & refactor Dockerfile to pass different Python versions and sccache bucket via build args (#7705)

Signed-off-by: kevin <kevin@anyscale.com>

lsy323 committed:
[Misc] fix typo in triton import warning (#7794)

joerunde committed:
[Frontend] error suppression cleanup (#7786)

Signed-off-by: Joe Runde <Joseph.Runde@ibm.com>

rkooo567 committed:
[Ray backend] Better error when pg topology is bad. (#7584)

Co-authored-by: youkaichao <youkaichao@126.com>

jikunshang committed:
[Hardware][Intel GPU] refactor xpu_model_runner for tp (#7712)

DamonFool committed:
[misc] Add Torch profiler support for CPU-only devices (#7806)

maxdebayser committed:
[BugFix] Fix server crash on empty prompt (#7746)

Signed-off-by: Max de Bayser <mbayser@br.ibm.com>

youkaichao committed:
[github][misc] promote asking llm first (#7809)

dsikka committed:
[Misc] Update `marlin` to use vLLMParameters (#7803)

simon-mo committed:
Bump version to v0.5.5 (#7823)

edwardzjl commented on vllm/entrypoints/chat_utils.py:
```diff
@@ -151,23 +230,38 @@ def _parse_chat_message_content_parts(
+                raise NotImplementedError(
+                    "Multiple 'table' input is currently not supported.")
+
+            table_data:List[Table] = cast(ChatCompletionContentPartTableParam,
```
Space after the colon.

edwardzjl commented on vllm/entrypoints/chat_utils.py:
```diff
@@ -122,15 +147,69 @@ def _get_full_image_text_prompt(image_token_str: str, text_prompt: str) -> str:
+    mm_futures: List[Union[Awaitable[MultiModalDataDict],
+                           MultiModalDataDict]] = []
 
+    is_table = False
```
'tabular_part'

edwardzjl commented on vllm/envs.py:
```diff
@@ -341,6 +348,23 @@ def get_default_config_root():
+    
+    
+    # MAX column for vllm table encoder
+    # Default is 30 seconds
```
30 seconds?

edwardzjl commented on vllm/multimodal/utils.py:
```diff
@@ -1,7 +1,6 @@
 import base64
 from io import BytesIO
 from typing import Union
-
```
Leave this line untouched?

Commit:
feat: support for `TableGPT` multimodal model
